x/ibc/02-client/keeper/keeper.go

```diff
@@ -96,6 +95,27 @@ func (k Keeper) SetClientConsensusState(ctx sdk.Context, clientID string, height
 	store.Set(ibctypes.KeyConsensusState(clientID, height), bz)
 }
 
+// GetLatestClientConsensusState gets the latest ConsensusState stored for a given client
+func (k Keeper) GetLatestClientConsensusState(ctx sdk.Context, clientID string) (exported.ConsensusState, bool) {
+	clientState, ok := k.GetClientState(ctx, clientID)
+	if !ok {
+		return nil, false
+	}
+	return k.GetClientConsensusState(ctx, clientID, clientState.GetLatestHeight())
+}
+
+// GetClientConsensusStatelTE will get the latest ConsensusState of a particular client at the latest height
+// less than or equal to the given height
+func (k Keeper) GetClientConsensusStateLTE(ctx sdk.Context, clientID string, maxHeight uint64) (exported.ConsensusState, bool) {
+	for i := maxHeight; i > 0; i-- {
+		consState, ok := k.GetClientConsensusState(ctx, clientID, i)
```

Comment:
This seems potentially less efficient than it could be, should we use `kvstore.Has` to check for each key?

Reply:
Yes did the most obvious thing to see if it all worked. Can optimize with `Has` now.

Could also use the `iavl.ReverseIterator` but i couldn't figure out how to get it to work for the old `Committer` logic so i think we can kick the can down the road on that

Reply:
Aye, that's fine.

Reply:
Add `TODO`